cql3/statements/describe_statement.cc

```diff
@@ -282,7 +282,9 @@ future<std::vector<description>> table(const data_dictionary::database& db, cons
 }
 
 future<std::vector<description>> tables(const data_dictionary::database& db, const lw_shared_ptr<keyspace_metadata>& ks, std::optional<bool> with_internals = std::nullopt) {
-    auto tables = ks->tables();
+    auto tables = boost::copy_range<std::vector<schema_ptr>>(ks->tables() | boost::adaptors::filtered([] (const schema_ptr& s) {
+        return !cdc::is_log_name(s->cf_name());
```

Comment:
You should use `cdc::is_log_for_some_table` instead. AFAIR it is legal to just create a table with the `_scylla_cdc_log` suffix if it doesn't exist - it doesn't magically become an actual CDC table. `cdc:: is_log_name` does not account for this while `cdc::is_log_for_some_table` does.

Reply:
I wonder what happens if you create a table xyz_scylla_cdc_log and only later try to create a table xyz with CDC enabled. Will it work? The secondary index code has a similar problem when it needs to name the backing materialized view, and it has some looping logic trying other fallback names if the name it wants to use is already taken. Does CDC have such things too? In any case, if we have a function `is_log_for_some_table()` we should definitely use that because it should take all these things into account (and if it doesn't, it's a pre-existing bug).

Reply:
I checked, and the existing code has a check_that_cdc_log_table_does_not_exist() check that will fail to create a table with CDC enabled (or to enable CDC) if the log name is already taken. There's no way for the code to pick a different name. And I guess it's fine - it's an esoteric case we need to handle gracefully but don't need to have a perfect solution for this case.

Reply:
> There's no way for the code to pick a different name. And I guess it's fine - it's an esoteric case we need to handle gracefully but don't need to have a perfect solution for this case.

I think this actually is a feature. We designed CDC so that users can consume the cdc tables directly just via a CQL driver (although we have some higher-level libraries for this). Making sure that the name is always `{base_table_name}_scylla_cdc_log` makes life earier for the users.

Reply:
I agree. In theory, the driver could have inspected some system table to figure out which table is which other table's log, but this is probably going over the top and the current approach is easier, and is fine. Sorry for taking over your comment for this discussion. You original comment, requesting to use is_log_for_some_table(), looks correct.